### PR TITLE
Add README, simplify some logic, make blank line separator easier to configure alongside objectMode

### PR DESCRIPTION
**Pull request by natevw**
Here's a variety of cleanup, mostly focused on getting clean single-line chunks in objectMode. Found this package today since it had the name I was gonna use if I couldn't find a nice simple "split a stream into lines" implementation already. So I figured I'd just run with yours ;-) Also added a README for good measure.

If this looks good to you, perhaps you could bump the package version and republish? (It should be fully backwards compatible, but might be worth a 0.2.0 since it's more than/not a bugfix.)

### CODE REVIEW

**Comment by natevw** [comment]
Any feedback on this, @jswartwood? I'd love to have these features usable via simple `npm install linefeed` instead my branch link, if you think they're useful to this module. Or if not, please let me know and I can maintain my changes under a different name.

**Comment by jswartwood** [comment]
sorry @natevw totally missed this PR. I'll do a review in a bit.

File: README.md [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+
+Besides the usual [stream.Transform](http://nodejs.org/api/stream.html#stream_class_stream_transform) configuration, the only other option is `{newline:string}`. Set it to any desired newline separator, including the empty string (which is especially useful with an object mode transform).
+
+
+## CLI
+
+There is also a minimal shell script you can pipe through if installed globally (or locally, if "./node_modules/.bin/" is in your path):
+
+```sh
+cat README.md | linefeed > README.unix_endings.txt
```
Comment (jswartwood) [inline README.md:40]: I like this example. Using it as a line-ending conversion utility is probably the most realistic cli use. Originally this whole module was created as a supplement for a talk I gave on Node streams, so the CLI only existed to provide a clear correlation to Unix cli philosophy. 

Takeaway... The bin should add a flag to specify a `--newline` character.

File: index.js [hunk 2]
```diff
@@ -9,26 +9,21 @@ function LineStream( opts ) {
...
-			this.push(this.newline);
-		} else {
-			this._temp += lines[i];
-		}
+
+	lines[0] = this._temp + lines[0]
+	for (var i = 0; i < last; i++) {
+		this.push(lines[i]+this.newline);
 	}
+	this._temp = lines[last];
```
Comment (jswartwood) [inline index.js:26]: This loop is so much cleaner... I don't know what I was thinking originally.

File: README.md (hunk 1 above)
Comment (natevw) [inline README.md:40]: Ah, okay! I wondered what you were using that for but for completeness figured I should include it in the docs (as far as I could understand reading the code).

**Comment by natevw** [comment]
Hey, thanks for merging and sorry to bother again, but could you also publish these as an update on npm then?

**Comment by jswartwood** [comment]
Absolutely. Thanks for the reminder.

**Comment by jswartwood** [comment]
Apologies that it took so long, but it's updated in npm now. Thanks again for the PR.

**Pull request merged**